Source/Particles/Collision/BinaryCollision/VirtualPhotonCreation.H

```diff
@@ -0,0 +1,244 @@
+
+                // Array with the PIDs of the virtual photons
+                uint64_t * AMREX_RESTRICT pid_vp = soa_vp.GetIdCPUData().data();
+
+                // Pointers to the arrays that will contain the particle attributes of the virtual photons
+                amrex::GpuArray<amrex::ParticleReal*,PIdx::nattribs> pa_vp;
+                for (int ia = 0; ia < PIdx::nattribs; ++ia) {
+                    pa_vp[ia] = soa_vp.GetRealData(ia).data();
+                }
+
+                // Second pass: populate the virtual photon species
+                amrex::ParallelForRNG (num,
+                [=] AMREX_GPU_DEVICE (amrex::Long i,  amrex::RandomEngine const& engine) noexcept
+                {
+                    // Primary particle
+                    amrex::ParticleReal ux_primary = soa.m_rdata[PIdx::ux][i];
+                    amrex::ParticleReal uy_primary = soa.m_rdata[PIdx::uy][i];
+                    amrex::ParticleReal uz_primary = soa.m_rdata[PIdx::uz][i];
+                    amrex::ParticleReal u_primary = std::sqrt(ux_primary*ux_primary + uy_primary*uy_primary + uz_primary*uz_primary);
+                    amrex::ParticleReal nx = ux_primary / u_primary; // normalized ux
+                    amrex::ParticleReal ny = uy_primary / u_primary; // normalized uy
+                    amrex::ParticleReal nz = uz_primary / u_primary; // normalized uz
+                    amrex::ParticleReal gamma_primary = std::sqrt( 1.0_rt + (ux_primary*ux_primary + uy_primary*uy_primary + uz_primary*uz_primary)*inv_c2 );
+
+#if defined (WARPX_DIM_3D)
+                    amrex::ParticleReal x  = soa.m_rdata[PIdx::x][i];
+                    amrex::ParticleReal y  = soa.m_rdata[PIdx::y][i];
+                    amrex::ParticleReal z  = soa.m_rdata[PIdx::z][i];
+#elif defined (WARPX_DIM_XZ)
+                    amrex::ParticleReal x  = soa.m_rdata[PIdx::x][i];
+                    amrex::ParticleReal z  = soa.m_rdata[PIdx::z][i];
+#elif defined (WARPX_DIM_RZ)
+                    amrex::ParticleReal x  = soa.m_rdata[PIdx::x][i];
+                    amrex::ParticleReal z  = soa.m_rdata[PIdx::z][i];
+                    amrex::ParticleReal theta  = soa.m_rdata[PIdx::theta][i];
+#elif defined (WARPX_DIM_1D_Z)
+                    amrex::ParticleReal z  = soa.m_rdata[PIdx::z][i];
+#elif defined (WARPX_DIM_RCYLINDER)
+                    amrex::ParticleReal x  = soa.m_rdata[PIdx::x][i];
+                    amrex::ParticleReal theta  = soa.m_rdata[PIdx::theta][i];
+#elif defined(WARPX_DIM_RSPHERE)
+                    amrex::ParticleReal x  = soa.m_rdata[PIdx::x][i];
+                    amrex::ParticleReal theta  = soa.m_rdata[PIdx::theta][i];
+                    amrex::ParticleReal phi  = soa.m_rdata[PIdx::phi][i];
+#endif
+                    amrex::ParticleReal w  = soa.m_rdata[PIdx::w][i];
+                    uint64_t pid = soa.m_idcpu[i]; // TODO: use this to update the parent lepton if needed
```

Comment:
Is this TODO comment still needed? If this has been done, you can remove the comment.